src/sas/qtgui/Perspectives/Fitting/FittingPerspective.py

```diff
@@ -187,6 +188,33 @@ def updateFromParameters(self, parameters):
         """
         self.currentTab.createPageForParameters(parameters)
 
+    def updateFromConstraints(self, constraint_dict):
+        """
+        Updates all tabs with constraints present in *constraint_dict*, where
+        *constraint_dict*  keys are the fit page name, and the value is a
+        list of constraints. A constraint is represented by a list [value,
+        param, value_ex, validate, function] of attributes of a Constraint
+        object
+        """
+        for fit_page_name, constraint_list in constraint_dict.items():
+            tab = self.getTabByName(fit_page_name)
+            for constraint_param in constraint_list:
+                if constraint_param is not None and len(constraint_param) == 5:
+                    value = constraint_param[0]
+                    param = constraint_param[1]
+                    value_ex = constraint_param[2]
+                    validate = constraint_param[3]
+                    function = constraint_param[4]
+                    constraint = Constraint()
+                    constraint.value = value
+                    constraint.func = function
+                    constraint.param = param
+                    constraint.value_ex = value_ex
+                    constraint.validate = validate
```

Comment:
we can probably leave out creating intermediate variables and assign list values directly to constraint attributes
`constraint.value = constraint_param[0]` etc.